trl/trainer/dpo_trainer.py

```diff
@@ -69,6 +66,136 @@
     import deepspeed
 
 
+def _tokenize(
+    example: Dict[str, Any],
+    tokenizer: "PreTrainedTokenizerBase",
+    is_vision_model: bool,
+) -> Dict[str, Any]:
+    """Tokenize a single example for DPO training."""
+
+    def tokenize_text(text: str, images: Union[List[str], torch.Tensor] = None) -> Dict[str, List[int]]:
+        if is_vision_model:
+            if "add_special_tokens" in inspect.signature(tokenizer).parameters:
+                processor_kwargs = {"add_special_tokens": False}
+            else:
+                processor_kwargs = {}
+            tokenized = tokenizer(text, images=images, **processor_kwargs)
+            tokenized = {k: v[0] for k, v in tokenized.items()}  # Unbatch
+            if not isinstance(tokenized["input_ids"], list):  # llava processor returns tensors
+                tokenized["input_ids"] = tokenized["input_ids"].tolist()
+                tokenized["attention_mask"] = tokenized["attention_mask"].tolist()
+        else:
+            tokenized = tokenizer(text, add_special_tokens=False)
+
+        return tokenized
+
+    prompt_tokens = tokenize_text(example["prompt"])
+    chosen_tokens = tokenize_text(example["prompt"] + example["chosen"], example.get("images"))
```

Comment:
It concatenates in the batch dim, while we want to concat each elements in the batch 

```python
prompt = ["My name is", "I live in"]
chosen = ["Quentin", "Lyon"]
print(prompt + chosen)
# ["My name is", "I live in", "Quentin", "Lyon"]
```